test/avatar.test.ts

```diff
@@ -0,0 +1,63 @@
+import { html, fixture, expect } from '@open-wc/testing';
+import { IgcAvatarComponent } from '../src/components/avatar/avatar';
+import '../igniteui-webcomponents.js';
+
+describe('Avatar', () => {
+  it('passes the a11y audit', async () => {
+    const el = await fixture<IgcAvatarComponent>(
+      html`<igc-avatar></igc-avatar>`
+    );
+
+    expect(el).shadowDom.to.be.accessible();
+  });
+
+  it('should initialize avatar component with default values', async () => {
+    const el = await fixture<IgcAvatarComponent>(
+      html`<igc-avatar></igc-avatar>`
+    );
+
+    expect(el.size).to.equal('small');
+    expect(el.shape).to.equal('square');
+  });
+
+  it('should set avatar size correctly', async () => {
+    const el = await fixture<IgcAvatarComponent>(
+      html`<igc-avatar size="small"></igc-avatar>`
+    );
+
+    expect(el.size).to.equal('small');
+    el.setAttribute('size', 'medium');
+    expect(el).dom.to.equal(`<igc-avatar size="medium"></igc-avatar>`);
```

Comment:
You should be checking the instance value of size, rather than whether the setAttribute works as expected.